ui/components/filter-list.jsx

```diff
@@ -7,79 +7,93 @@ import React from 'react';
-  keywordIds: [],
-  keyword: {},
-  query: {},
+  existingIds: [],
+  idToRemove: 0,
+  location: { pathname: '', query: {} },
+  name: '',
+  removeParam: '',
 };
 Filter.propTypes = {
-  keywordIds: React.PropTypes.arrayOf(React.PropTypes.string),
-  keyword: React.PropTypes.shape({
-    id: React.PropTypes.number,
-    name: React.PropTypes.string,
+  existingIds: React.PropTypes.arrayOf(React.PropTypes.number),
+  idToRemove: React.PropTypes.number,
+  location: React.PropTypes.shape({
+    pathname: React.PropTypes.string,
+    query: React.PropTypes.shape({}),
   }),
-  query: React.PropTypes.shape({}),
+  name: React.PropTypes.string,
+  removeParam: React.PropTypes.string,
+};
+
+/* Mapping between a lookup type (e.g. "keywords") and the query field that
+ * this would need to modify to add/remove lookup values */
+export const searchParam = {
+  keywords: 'keywords__id__in',
+  policies: 'policy_id__in',
 };
 
-export default function FilterList({ keywords, router }) {
-  const { location: { query } } = router;
-  const keywordIds = (query.keywords__id__in || '').split(',');
-  const removeQuery = Object.assign({}, query);
-  delete removeQuery.page;
-  delete removeQuery.keywords__id__in;
+export default function FilterList({ existingFilters, lookup, router }) {
+  const filterIds = existingFilters.map(existing => existing.id);
   return (
     <div className="req-filter-ui">
-      <h3>Keywords</h3>
+      <h3>{lookup.charAt(0).toUpperCase() + lookup.substr(1)}</h3>
       <ol className="list-reset">
-        { keywords.map(keyword =>
+        { existingFilters.map(filter =>
           <Filter
-            key={keyword.id} keywordIds={keywordIds} keyword={keyword}
-            query={removeQuery}
+            key={filter.id} existingIds={filterIds} idToRemove={filter.id}
+            location={router.location} name={filter[apiParam[lookup]]}
+            removeParam={searchParam[lookup]}
           />)}
       </ol>
-      <SearchAutocomplete lookup="keywords" insertParam="keywords__id__in" router={router} />
+      <SearchAutocomplete lookup={lookup} insertParam={searchParam[lookup]} router={router} />
     </div>
   );
 }
 FilterList.defaultProps = {
-  keywords: [],
-  router: { location: { query: {} } },
+  existingFilters: [],
+  lookup: 'keywords',
+  router: { location: {} },
 };
 FilterList.propTypes = {
-  keywords: React.PropTypes.arrayOf(React.PropTypes.shape({
+  existingFilters: React.PropTypes.arrayOf(React.PropTypes.shape({
     id: React.PropTypes.number,
-    name: React.PropTypes.string,
   })),
+  lookup: React.PropTypes.oneOf(Object.keys(searchParam)),
   router: React.PropTypes.shape({
-    location: React.PropTypes.shape({
-      query: React.PropTypes.shape({
-        keywords__id__in: React.PropTypes.string,
-      }),
-    }),
+    location: React.PropTypes.shape({}),
   }),
 };
 
-export function fetchData({ location: { query } }) {
+export function fetchKeywords({ location: { query } }) {
```

Comment:
Do you think it'd make sense to encapsulate all of the API interactions into one module/class at this point? I've seen that in several of these frontend projects, but I don't have a strong feeling either way.